include/zephyr/kernel.h

```diff
@@ -958,6 +958,22 @@ __syscall void k_thread_priority_set(k_tid_t thread, int prio);
  *
  */
 __syscall void k_thread_deadline_set(k_tid_t thread, int deadline);
+
+/**
+ * @brief Get the absolute deadline of a thread.
+ *
+ * This routine gets the deadline of a thread to be used by the EDF
+ * scheduler when deciding which thread has the highest priority.
+ *
+ * @note You should enable @kconfig{CONFIG_SCHED_DEADLINE} in your project
+ * configuration.
+ * 
+ * @param thread Thread from which to get the deadline
+ * @return absolute deadline value, in cycle units
+ *
+ */
+int32_t k_thread_deadline_get(k_tid_t thread);
```

Comment:
What's the use case you're imagining here? Basically I have a hard time imagining a use case for a function like this that isn't a huge race conditions factory.  If you just need it for the sample, you can just as easily compute it (subject to the same raciness!) from the time of the original deadline call, no?

Also: the returned value is an absolute cycle time e.g. as you might see from k_cycle_get_32(), and not a delta from current time as passed to k_thread_deadline_set().  That asymmetry is a footgun and should be called out in the docs.

Reply:
I'm sorry, why would this cause a race condition? It's just a getter, and deadlines are not expected to change so often anyway.

The use case is... if the application code need to know what deadline value was set. If this function does not exist, the application will have to access `thread->base.prio_deadline` and that is not good practice AFAIK or even consistent with the API anyway. Just like `k_thread_name_get` is a wrapper for `thread->name` and `k_thread_priority_get` is a wrapper for `thread->priority`.